web_src/js/utils/dom.ts

```diff
@@ -340,3 +347,11 @@ export function querySingleVisibleElem<T extends HTMLElement>(parent: Element, s
   if (candidates.length > 1) throw new Error(`Expected exactly one visible element matching selector "${selector}", but found ${candidates.length}`);
   return candidates.length ? candidates[0] as T : null;
 }
+
+export function addElemsEventListener<T extends HTMLElement>(parent: Node, type: string, selector: string, listener: (elem: T, e: Event) => void | Promise<any>, options?: boolean | AddEventListenerOptions) {
```

Comment:
To reviewers: this function helps migrating jQuery `$(document).on('click', function() { this.xxx } )` to new code.

jQuery's `this` is `closest(selector)`, it is not `e.target`, nor `e.currentTarget`.

Reply:
maybe call it `addDelegatedEventListener`?

Reply:
done